Fabric.Identity.API/Services/ClaimsService.cs

```diff
@@ -0,0 +1,147 @@
+﻿using Fabric.Identity.API.Configuration;
+using Fabric.Identity.API.Exceptions;
+using Fabric.Identity.API.Models;
+using Fabric.Identity.API.Quickstart.Account;
+using IdentityModel;
+using IdentityServer4.Models;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http.Authentication;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Security.Claims;
+
+namespace Fabric.Identity.API.Services
+{
+    public class ClaimsService
+    {
+        private readonly IAppConfiguration _appConfiguration;
+
+        public ClaimsService(
+            IAppConfiguration appConfiguration)
+        {
+            _appConfiguration = appConfiguration;
+        }
+
+        public ClaimsResult GenerateClaimsForIdentity(List<Claim> claims, AuthenticateInfo info, AuthorizationRequest context, string schemaItem)
+        {
+            var userIdClaim = this.GetUserIdClaim(claims);
+            var provider = info.Properties.Items["scheme"];
+            var result = new ClaimsResult()
+            {
+                ClientId = context?.ClientId,
+                UserId = userIdClaim.Value,
+                Provider = provider
+            };
+
+            if (this.IsAzureEnabled(schemaItem))
+            {
+                if (this.ValidateAzureAD(claims))
+                {
+                    var issuerClaim = claims.FirstOrDefault(x => x.Type == JwtClaimTypes.Issuer);

```

Comment:
Why would we throw an exception if the issuer claim is valid and present? I think we might just be missing a `!` operator on this check?

Reply:
So if you get into here, the AzureAD issuer claim is not valid, Based on the function above.  So we are getting the issuer that is being used to help diagnose the issue.